pkg/kp/rcstore/rc.go

```diff
@@ -0,0 +1,95 @@
+package rcstore
+
+import (
+	"github.com/square/p2/pkg/labels"
+	"github.com/square/p2/pkg/pods"
+)
+
+type ReplicationControllerDesires struct {
+	replicasDesired int
+	disabled        bool
+}
+
+type ReplicationControllerData struct {
+	// It is expected that these fields will never be changed after an RC is created.
+	id           string
+	manifest     *pods.Manifest
+	nodeSelector labels.Selector
+	podLabels    labels.Set
+
+	watcher desireWatcher
+
+	// It is expected that these fields can be changed.
+	ReplicationControllerDesires
```

Comment:
That's true. I did this so that the internal `watch` function on `desireWatcher` would only need to return those fields, since those are the only ones expected to change. Now that I think about it though, I could just make the `desireWatcher` modify in place instead of returning the fields. In that case `ReplicationControllerDesires` could just go away.

I may even structure the console keys in a way such that these keys live under e.g. `replication_controllers/some_id_here/desires/` and I just watch that instead of `replication_controllers/some_id_here`.

I've also realized `ReplicationControllerDesires` isn't even referenced anywhere outside the `rcstore` package (everything else is using exported methods on `ReplicationControllerData`), which means I could just not export it.